libbeat/cmd/instance/beat.go

```diff
@@ -318,6 +321,41 @@ func (b *Beat) launch(bt beat.Creator) error {
 	return beater.Run(&b.Beat)
 }
 
+// checkXPackMonitoring checks if xpack is enabled and collection is enabled
+func (b *Beat) checkXPackMonitoring() bool {
+
+	if b.Config.Output.Name() != "elasticsearch" {
+		logp.Info("Disabling X-Pack Monitoring as output is not Elasticsearch.")
+		return false
+	}
+
+	esConfig := b.Config.Output.Config()
+	esClient, err := elasticsearch.NewConnectedClient(esConfig)
+	if err != nil {
+		logp.Info("Disabling X-Pack monitoring as connection to Elasticsearch could not be established. Error: %s", err)
+		return false
+	}
+	// Check if monitoring x-pack feature is enabled
+	_, body, err := esClient.Request("GET", "/_xpack", "", nil, nil)
+	features := struct {
+		Features struct {
+			Monitoring struct {
+				Enabled bool `json:"enabled"`
+			} `json:"monitoring"`
+		} `json:"features"`
+	}{}
+	err = json.Unmarshal(body, &features)
+	if err != nil {
+		logp.Debug("monitoring", "Error fetching monitoring enabled feature from Elasticsearch: %s", err)
```

Comment:
Shouln't this be logged at error level?

Reply:
The idea here was that in all cases where something strange comes back from ES because of Monitoring, not errors are logged. When reading this review I realised there should be an additional error check on line 339 for the error there.